Dockerfile

```diff
@@ -0,0 +1,46 @@
+ARG PYTHON_VERSION
+FROM python:${PYTHON_VERSION} AS base
```

Comment:
Letting the user vary the base image is a substantial change and potentially complicates your later steps - what if the user decides they want to use a Windows based python image? Is there any significant advantage in having it be variable at this stage?